sinabs/backend/dynapcnn/dynapcnnnetwork_module.py

```diff
@@ -0,0 +1,335 @@
+            with at least one exit destination. Values are list of dicts, providing
+            for each exit destination the negative valued ID ('destination_id'),
+            the index of that destination within the list of destinations of the
+            corresponding `DynapcnnLayer` ('destination_index'), and the pooling
+            for this destination.
+        """
+        exit_layers = dict()
+        for layer_idx, destinations in self.destination_map.items():
+            exit_destinations = []
+            for i, dest in enumerate(destinations):
+                if dest < 0:
+                    exit_destinations.append(
+                        {
+                            "destination_id": dest,
+                            "destination_index": i,
+                            "pooling": self.dynapcnn_layers[layer_idx].pool[i],
+                        }
+                    )
+            if exit_destinations:
+                exit_layers[layer_idx] = exit_destinations
+
+        return exit_layers
+
+        
+    def setup_dynapcnnlayer_graph(self, index_layers_topologically: bool = False):
+        """ Set up data structures to run forward pass through dynapcnn layers
+        
+        Parameters
+        ----------
+        - index_layers_topologically (bool): If True, will assign new indices to
+            dynapcnn layers such that they match their topological order within the
+            network graph. This is not necessary but can help understand the network
+            more easily when inspecting it.
+        """
+        self._dynapcnnlayer_edges = self.get_dynapcnnlayers_edges()
+        self.add_entry_points_edges(self._dynapcnnlayer_edges)
+        self._sorted_nodes = topological_sorting(self._dynapcnnlayer_edges)
+        self._node_source_map = self.get_node_source_map(self._dynapcnnlayer_edges)
+        if index_layers_topologically:
+            self.reindex_layers(self._sorted_nodes)
+
+    def get_dynapcnnlayers_edges(self) -> Set[Edge]:
+        """Create edges representing connections between `DynapcnnLayer` instances.
+
+        Returns
+        ----------
+        - dcnnl_edges: a set of edges using the IDs of `DynapcnnLayer` instances. These edges describe the computational
+            graph implemented by the layers of the model (i.e., how the `DynapcnnLayer` instances address each other).
+        """
+        dcnnl_edges = set()
+
+        for dcnnl_idx, destination_indices in self._destination_map.items():
+            for dest in destination_indices:
+                if dest >= 0:  # Ignore negative destinations (network exit points)
+                    dcnnl_edges.add((dcnnl_idx, dest))
+
+        return dcnnl_edges
+
+    def add_entry_points_edges(self, dcnnl_edges: Set[Edge]):
+        """Add extra edges `('input', X)` to `dcnnl_edges` for
+        layers which are entry points of the `DynapcnnNetwork`, i.e.
+        `handler.entry_node = True`.
```

Comment:
DynapcnnLayer handlers are not used anymore - docstring needs update.

Reply:
double-check.